### ISSUE TITLE
Support generating markdown with mermaid

### ISSUE THREAD

**Issue by xenoterracide** [spring-projects/spring-modulith#844]
Github supports rendering [Mermaid](https://mermaid.js.org) (an alternative to plantuml) in its markdown (no idea on asciidoc). So does microsofts azure thing.

Feels like a nice to have built in.

**Comment by odrotbohm** [comment]
Can you clarify what exactly you're looking for here? Is it generating diagrams in Mermaid VS PlantUML? Or is it generating Application Module Canvases in Markdown VS. Asciidoctor?

I'm a bit hesitant on the latter, as the support for tables in Markdown is virtually nonexistent (it's primarily using HTML tables). Furthermore, the Asciidoctor output allows transforming it into other markup formats if required.

For the former, I guess we'd have to investigate what the C4 exporter supports. That said, given that PlantUML is so ubiquitously supported that I currently don't see much need for any other format.

**Comment by xenoterracide** [comment]
>  PlantUML is so ubiquitously supported

I haven't found that anything really supports it at all. It's only supported by virtue that you are generating images, and even things that support it directly have done so really poorly resulting in ugly output. I can only think of a single app I saw online that supported it. I can't remember its name, one of those live collaboration drawing board tools. It didn't apply any styling to make it not ugly. 

This is not to say that it's *unusable*, I actually like it, and to some extent prefer it.

> Is it generating diagrams in Mermaid VS PlantUML?

let's just say mermaid.

> Application Module Canvases in Markdown VS. Asciidoctor

Mostly I think I'm thinking about wanting to stick it in places like my README on github, even if that's copypasta, and it'd probably have to be. I'm just clarifying the use case. Mermaid with whatever githubs limited subset is would be sufficient. I think microsofts subset of mermaid is even more limited... basically you don't have to support *all* of mermaid ;)

**Comment by sw-tracker** [comment]
Hi there,

We publish all of our documentation in backstage (which uses mkdocs under the hood). For that we would also like to have support for markdown (in fact there are many markdown to html tools so I think this request adds a lot of value). I know there is a plugin to add asciidoc support to mkdocs, but having to request extra plugins being installed in the company just because of spring modulith might not fly.

Additionally, using plantUML with backstage is quite cumbersome at the moment. We are about to upgrade so maybe this is better in future versions. But for now, all devs are using mermaid because that works really easily. In general, Mermaid is widely used so the support would also add value.

In summary, I think generating markdown is a must.

Thanks
